blog-posts/event-sourcing-serveless/event-sourcing-serveless.md

```diff
@@ -0,0 +1,171 @@
+
+We have the two sides of the CQRS architecture, with the writing in the command and the reading in the query.
+
+<img src="./assets/archi-step-5-query.png" style="display: block; margin: auto" width="1000"/>
+
+But there is a big part missing here. How do we transform data from events to projections? How is the synchronization between read and write models ensured?
+
+## Data synchronization
+
+The pain point in CQRS is to have reliable reading data compared to the source of truth. We need to find a way to compute and write projections according to events written in the event store.
+
+### Projector
+
+For the writing part, we have already done the job in the previous development. We will simply use lambda functions to put data in DynamoDB. These lambda are called projectors.
+
+But where does the data come from? It's legitimate to think about Event Bridge events to carry data to trigger lambda. For example, the total balance projector can be triggered by a debit event of 20$ `{ 'eventType': 'Debit', 'amount': 20}`. The DynamoDB UpdateItem method allows us to manipulate the value of an existing record without querying it beforehand. In the case of a debit we can simply subtract 20 to the existing value of the projection. The same projector can listen to credit event also and update the projection the same way. We can think of updating the last transaction with an event carrying `{ 'eventType': 'Credit', 'amount': 40}` by overwriting the previous value of the projection.
+
+<img src="./assets/archi-step-6-projector.png" style="display: block; margin: auto" width="1200"/>
+
+We still don't know how are created these events carrying the data that is written in the event store.
+
+### DynamoDB streams and fanout lambda
+
+We want to react in real time to events written inside the event store. For that DynamoDb gives us a powerful tool that is called DynamoDB streams. We plug a lambda to the streams that will be triggered by every action made in the database. DynamoDB batches its streams of events in arrays, so we filter the `INSERT` actions and for each of them we publish an event in Amazon EventBridge.
+
+> :question: You may wonder why not directly plug projector to the streams
+
+For 3 reasons:
+
+- A lambda that is plugged to the streams and that fails will retry indefinitely
+- In every projector you would have to handle the filtering of events, and every projector would be triggered by every event
```

Comment:
Je comprends pas ta phrase -> la dernière raison - la limitation à 2 lambda - est la raison pour laquelle on ne ship pas les projecteurs directement sur DynamoDB streams, mais tu peux filtrer comme sur évent bridge -> http://serverless.com//blog/guest-post-developing-a-serverless-scheduler-using-dynamodb-ttl-and-filtered-streams